log/keyvalue.go

```diff
@@ -256,6 +257,14 @@ func (v Value) Equal(w Value) bool {
 	case KindSlice:
 		return slices.EqualFunc(v.asSlice(), w.asSlice(), Value.Equal)
 	case KindMap:
+		sv := v.asMap()
+		slices.SortFunc(sv, func(a, b KeyValue) int {
+			return cmp.Compare(a.Key, b.Key)
+		})
+		sw := w.asMap()
+		slices.SortFunc(sw, func(a, b KeyValue) int {
+			return cmp.Compare(a.Key, b.Key)
+		})
 		return slices.EqualFunc(v.asMap(), w.asMap(), KeyValue.Equal)
```

Comment:
This mutates the underlying slice.

I know it's more costly to clone the slice, but I think it is appropriate.  It is fine for our uses, but this is a footgun waiting to happen if others start using it for comparison.